### DISCUSSION TITLE
Next.JS/React.JS - Style not apply if using variable as class name outside of component file

### DISCUSSION THREAD

**Discussion by ARiyou2000** [tailwindlabs/tailwindcss#11125] [Help] [answered]
**What version of Tailwind CSS are you using?**

"^3.3.2"

**What build tool (or framework if it abstracts the build tool) are you using?**

next: "13.3.0", react: "18.2.0", postcss: "8.4.21"

**What version of Node.js are you using?**

v18.16.0

**What browser are you using?**

Chrome

**What operating system are you using?**

Ubuntu 20.4

**Reproduction URL**

It is not reproducible with Tailwind Playgroung, but here is a temp link to get a gist of it:
[GitHub](https://github.com/ARiyou2000/tailwind-styleWithGlobalClass-issue)

**Describe your issue**

As the doc suggests I have created class variables for similar styles for my components.

> export const blurredCard = "rounded-[20px] backdrop-blur-[10px] bg-[#C2C2C266]";

But when importing it to use it with components, even though the class will apply (checked in Chroom devTool [F12]), styles will not apply.
More accurately only styles that have been mentioned in other components as inline class names will apply.

My guess is that the class name will apply as a string, but the relevant style will not be delivered in the bundle.

As far as I know, Tailwind and Webpack will try to keep bundle size as small as possible by omitting unused styles (styles that their respective class name has not mentioned)

**Comment by adamwathan** [comment, marked as answer]
Hey! As long as `TailwindCustomClass.js` is covered by your [content configuration](https://tailwindcss.com/docs/content-configuration#configuring-source-paths) everything will work as expected 👍

**Comment by asmin52** [comment]
I should say you have composed an extraordinary article. The manner in which you have depicted everything is exceptional.  
[Clover.com](https://www.cloverdashboard.net/)
